### ISSUE TITLE
`not-callable` false positive for `types.FunctionType`

### ISSUE THREAD

**Issue by Jackenmen** [pylint-dev/pylint#7500]
### Bug description

```py
# pylint: disable=missing-docstring

import types

code = compile(input(">>> "), "<stdin>", "exec")
types.FunctionType(code, {})()
```

A more complete example of public code constructing `types.FunctionCode` object can be found at:
https://github.com/python/cpython/blob/34de67c094776bdc719d87f8810e053a2acc915a/Lib/asyncio/__main__.py#L32

### Command used

```shell
pylint repro.py
```

### Pylint output

```shell
************* Module repro
repro.py:6:0: E1102: types.FunctionType(code, {}) is not callable (not-callable)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]
```

### OS / Environment

Kubuntu 22.04.1 LTS
